pkgs/development/haskell-modules/configuration-common.nix

```
@@ -238,10 +238,16 @@ self: super: {
   digit = doJailbreak super.digit;
 
   # 2020-06-05: HACK: does not pass own build suite - `dontCheck`
-  hnix = generateOptparseApplicativeCompletion "hnix" (dontCheck super.hnix);
+  # 2022-06-17: Use hnix-store 0.5 until hnix 0.17
+  hnix = generateOptparseApplicativeCompletion "hnix" (dontCheck (
+    super.hnix.override {
+      hnix-store-core = hnix_store_core_0_5_0_0;
+      hnix-store-remote = hnix_store_remote_0_5_0_0.override { hnix-store-core = hnix_store_core_0_5_0_0; };
+    }
```

You can likely write this a little simpler as

```suggestion
    super.hnix.overrideScope (hself: hsuper: {
      hnix-store-core = hself.hnix_store_core_0_5_0_0;
      hnix-store-remote = hself.hnix_store_remote_0_5_0_0;
    })
```

I didn't test this though.